doc/nrf/libraries/networking/lwm2m_client_utils.rst

```diff
@@ -71,6 +71,9 @@ But if this waiting period is longer than the maximum delay set for the data tra
 .. note::
    Connection pre-evaluation consumes a small amount of energy every time it requests information about a cell.
 
+The :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option enables DTLS Connection Identifier functionality in handshake.
+Enabling this feature requires that :kconfig:option:`CONFIG_LWM2M_TLS_SESSION_CACHING` is turned off and either :kconfig:option:`CONFIG_LWM2M_RD_CLIENT_STOP_POLLING_AT_IDLE` or :kconfig:option:`CONFIG_LWM2M_RD_CLIENT_LISTEN_AT_IDLE` is enabled.
```

Comment:
.rst states that it is required that `CONFIG_LWM2M_TLS_SESSION_CACHING` is disabled and kconfig states that it should be disabled when `CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` is enabled.
These should more clear in stating as the required feels more like must than should.

Also, if this is a must requirement, it could use a depends on in the kconfig.